insert_data_to_vector_db_weaviate/main.bal

```diff
@@ -0,0 +1,64 @@
+import ballerina/io;
+import ballerinax/googleapis.sheets;
+import ballerinax/openai.embeddings;
+import ballerinax/weaviate;
+
+configurable string openAIToken = ?;
+configurable string weaviateToken = ?;
+configurable string weaviateURL = ?;
+
+configurable string sheetAccessToken = ?;
+configurable string sheetId = ?;
+configurable string sheetName = ?;
+
+const RANGE = "A2:C630";
+const NO_OF_COLUMNS = 3;
+const className = "QuestionAnswerStore";
+const MODEL = "text-embedding-ada-002";
+
+final embeddings:Client openai = check new ({auth: {token: openAIToken}});
+final weaviate:Client weaviate = check new ({auth: {token: weaviateToken}, timeout: 100}, weaviateURL);
+final sheets:Client gsheets = check new ({auth: {token: sheetAccessToken}});
+
+public function main() returns error? {
+    sheets:Range range = check gsheets->getRange(sheetId, sheetName, RANGE);
+    string[] data = [];
+    weaviate:Object[] documentObjects = [];
+
+    // Iterate through the data stream and extract the content
+    foreach var row in range.values {
+        if (row.length() != NO_OF_COLUMNS) {
+            continue;
+        }
+        weaviate:Object obj = {
+            'class: className,
+            properties: {
+                "title": row[0],
+                "question": row[1],
+                "answer": row[2]
+            }
+        };
+        documentObjects.push(obj);
+        data.push(row[1].toString());
+    }
+    embeddings:CreateEmbeddingResponse embeddingResponse = check openai->/embeddings.post({model: MODEL, input: data});
+
+    // Insert embedding vectors to the Weaviate objects
+    foreach int i in 0 ... embeddingResponse.data.length() - 1 {
+        documentObjects[i].vector = embeddingResponse.data[i].embedding;
+    }
+
+    weaviate:ObjectsGetResponse[] responses = check weaviate->/batch/objects.post({objects: documentObjects});
+
+    string[] failedQuestions = [];
+    foreach var res in responses {
+        if res.result["errors"]["error"] != () {
+            failedQuestions.push(res.properties["question"].toString());
+        }
+    }
```

Comment:
Line 53:58 can be replaced as follows.

```ballerina
string[] failedQuestions = from var res in responses
        where res.result?.errors?.'error != ()
        select res.properties["question"].toString();
```

Reply:
Addressed in https://github.com/ballerina-guides/ai-samples/pull/60/commits/2bbad1aa77295f40a3191110f4502e51165dd282.